### ISSUE TITLE
Server "h3://xxx.com/dns-query": could not be used, please check that you've written it correctly

### ISSUE THREAD

**Issue by Mircc** [AdguardTeam/AdGuardHome#6375]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Linux, Synology DS220+ （DSM7.2) in Docker 

### Installation

GitHub releases or script from README

### Setup

Other (please mention in the description)

### AdGuard Home version

ver: v0.107.40

### Action

Upstream DNS servers no support h3 dns,tips:
Blocking mode
Default: Respond with zero IP address (0.0.0.0 for A; :: for AAAA) when blocked by Adblock-style rule; respond with the IP address specified in the rule when blocked by /etc/hosts-style rule
REFUSED: Respond with REFUSED code
NXDOMAIN: Respond with NXDOMAIN code
Null IP: Respond with zero IP address (0.0.0.0 for A; :: for AAAA)
Custom IP: Respond with a manually set IP address
Blocked response TTL
Specifies for how many seconds the clients should cache a filtered response
10
DNS cache configuration
Here you can configure DNS cache
Cache size
DNS cache size (in bytes). To disable caching, leave empty.
4194304
Override minimum TTL
Extend short time-to-live values (seconds) received from the upstream server when caching DNS responses.
Enter minimum TTL (seconds)
Override maximum TTL
Set a maximum time-to-live value (seconds) for entries in the DNS cache.
Enter maximum TTL (seconds)

Access settings
Here you can configure access rules for the AdGuard Home DNS server
Allowed clients
A list of CIDRs, IP addresses, or [ClientIDs](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients#clientid). If this list has entries, AdGuard Home will accept requests only from these clients.
Disallowed clients
A list of CIDRs, IP addresses, or [ClientIDs](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients#clientid). If this list has entries, AdGuard Home will drop requests from these clients. This field is ignored if there are entries in Allowed clients.
Disallowed domains
Not to be confused with filters. AdGuard Home drops DNS queries matching these domains, and these queries don't even appear in the query log. You can specify exact domain names, wildcards, or URL filter rules, e.g. "example.org", "*.example.org", or "||example.org^" correspondingly.
version.bind
id.server
hostname.bind
[Homepage](https://github.com/AdguardTeam/AdGuardHome)[Privacy Policy](https://link.adtidy.org/forward.html?action=privacy&from=ui&app=home)

English
Copyright © 2023 [AdGuard](https://link.adtidy.org/forward.html?action=home&from=ui&app=home)
Version: 
v0.107.40
Server "h3://dns.alidns.com/dns-query": could not be used, please check that you've written it correctly

### Expected result

I wish I could support http3 dns servers!
AdGuardHome.yaml config file has serve_http3: true and use_http3_upstreams: true turned on, and restarted the service several times, but it still doesn't work!

### Actual result

Server "h3://dns.alidns.com/dns-query": could not be used, please check that you've written it correctly

Screenshot of adguard dns configuration interface
https://raw.githubusercontent.com/Mircc/sgmodule/main/ad1.png
Screenshot of AdGuardHome.yaml Configuration
https://raw.githubusercontent.com/Mircc/sgmodule/main/ad2.png

adguardhome Deployed in the docker environment of a Synology DSM7.2 system

**Comment by ainar-g** [comment]
Most likely a duplicate of #6335.  Please check the latest Edge releases.
